src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleShell.cs

```diff
@@ -14,6 +14,24 @@ namespace Microsoft.PowerShell
     public static
     class ConsoleShell
     {
+        /// <summary>Entry point in to ConsoleShell. Used to create a custom Powershell console application</summary>
+        /// <param name="initialSessionState">InitialSessionState to be used by the ConsoleHost.</param>
+        /// <param name="bannerText">Banner text to be displayed by ConsoleHost.</param>
+        /// <param name="helpText">Help text for the shell.</param>
+        /// <param name="args">Commandline parameters specified by user.</param>
+        /// <returns>An integer value which should be used as exit code for the process.</returns>
+        public static int Start(InitialSessionState initialSessionState, string bannerText, string helpText, string[] args)
+        {
+            if (initialSessionState == null)
+            {
+                throw PSTraceSource.NewArgumentNullException(nameof(initialSessionState));
+            }
+
+            ConsoleHost.DefaultInitialSessionState = initialSessionState;
+
+            return Start(bannerText, helpText, args);
+        }
```

Comment:
I prefer to have the dependency other way around. The current public API `Start(string bannerText, string helpText, string[] args)` will fail because `ConsoleHost.DefaultInitialSessionState` is never set.
```
        public static int Start(string bannerText, string helpText, string[] args)
           : this(InitialSessionState.CreateDefault2(), bannerText, helpText, args)
        {
        }

        public static int Start(InitialSessionState initialSessionState, string bannerText, string helpText, string[] args)
        {
            if (initialSessionState == null)
            {
                throw PSTraceSource.NewArgumentNullException(nameof(initialSessionState));
            }

            if (args == null)
            {
                throw PSTraceSource.NewArgumentNullException(nameof(args));
            }

            ConsoleHost.DefaultInitialSessionState = initialSessionState;

            return ConsoleHost.Start(bannerText, helpText, args);
        }
```
Along with this change, we need to remove the `ConsoleHost.DefaultInitialSessionState = InitialSessionState.CreateDefault2();` line at https://github.com/PowerShell/PowerShell/blob/master/src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs#L66

Reply:
Changed.
As far as i know the ":" way doesn't work for methods but i have changed the essence to fit what you described.

Thanks!

Reply:
Ah, yes, I must have thought them as constructors :)